pkgs/applications/networking/instant-messengers/teamspeak/client.nix

```diff
@@ -97,34 +84,38 @@ stdenv.mkDerivation rec {
     runHook postBuild
   '';
 
-  installPhase =
-    ''
-      runHook preInstall
-
-      # Install files.
-      mkdir -p $out/lib/teamspeak
-      mv * $out/lib/teamspeak/
-
-      # Make a desktop item
-      mkdir -p $out/share/applications/ $out/share/icons/hicolor/64x64/apps/
-      cp ${pluginsdk}/doc/_static/logo.png $out/share/icons/hicolor/64x64/apps/teamspeak.png
-      cp ${desktopItem}/share/applications/* $out/share/applications/
-
-      # Make a symlink to the binary from bin.
-      mkdir -p $out/bin/
-      ln -s $out/lib/teamspeak/ts3client $out/bin/ts3client
-
-      runHook postInstall
-    '';
-
-  preFixup =
-    ''
-      wrapProgram $out/bin/ts3client \
-    '' # wayland is currently broken, remove when TS3 fixes that
-    + ''
-      --set QT_QPA_PLATFORM xcb \
-      --set NIX_REDIRECTS /usr/share/X11/xkb=${xkeyboard_config}/share/X11/xkb
-    '';
+  desktopItems = [
+    (makeDesktopItem {
+      name = "teamspeak";
+      exec = "ts3client";
+      icon = "teamspeak";
+      comment = "The TeamSpeak voice communication tool";
+      desktopName = "TeamSpeak";
+      genericName = "TeamSpeak";
+      categories = [ "Network" ];
+    })
+  ];
+
+  qtWrapperArgs = [
+    # wayland is currently broken, remove when TS3 fixes that
+    "--set QT_QPA_PLATFORM xcb"
+    "--set NIX_REDIRECTS /usr/share/X11/xkb=${xkeyboard_config}/share/X11/xkb"
```

Comment:
Didn't we remove the libredirect earlier?

Reply:
I was not aware that libredirect is a Nixpkgs-specific thing. The application appears to be working fine without it to me. How would this manifest?